tools/buildsys/src/cache.rs

```diff
@@ -52,26 +54,28 @@ impl LookasideCache {
             let tmp = PathBuf::from(format!(".{}", name));
 
             // first check the lookaside cache
-            let url = format!("{}/{}/{}/{}", LOOKASIDE_CACHE, name, hash, name);
+            let url = format!("{}/{}/{}/{}", lookaside_cache, name, hash, name);
             match Self::fetch_file(&url, &tmp, hash) {
                 Ok(_) => {
                     fs::rename(&tmp, path)
                         .context(error::ExternalFileRenameSnafu { path: &tmp })?;
                     continue;
                 }
                 Err(e) => {
-                    eprintln!("{}", e);
+                    // next check with upstream, if permitted
+                    if upstream_fallback {
```

Comment:
There are a few commits that change this line, will it be better to squash them or are you planning to squash them once the PR is merged?

Reply:
The commits feel logical to me :shrug:. They sort of deal with different concerns.